File: api/src/main/java/com/api/app/emum/MEM002.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum MEM002 implements CommonCode {
    EARN("001", "적립", 1, "", ""),
    USE("002", "사용", 2, "", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static MEM002 findByCode(String code) {
        return CommonCodeUtil.findByCode(MEM002.class, code);
    }
}

File: api/src/main/kotlin/com/api/app/emum/MEM002.kt
package com.api.app.emum

/** 포인트거래구분코드: pointTransactionCode */
enum class MEM002(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    EARN("001", "적립", 1, "", ""),
    USE("002", "사용", 2, "", ""),
    ;

    companion object {
        fun findByCode(code: String): MEM002? = CommonCodeUtil.findByCode(MEM002::class.java, code)
    }
}
